tests/apple-exporter/components/test-class-table.php

```diff
@@ -146,54 +146,92 @@ public function test_dark_colors() {
-		// Header Background, Header Text Color.
-		$this->assertTrue( isset( $json['componentStyles']['default-table']['tableStyle']['headerCells']['conditional'] ) );
+		// Ensure component level conditional is set.
+		$this->assertEquals(
+			'dark-table',
+			$json['components'][1]['conditional'][0]['style']
+		);
 
-		// Ensure Color Values match.
+		// Ensure border color values match.
 		$this->assertEquals(
 			'#abcdef',
-			$json['componentStyles']['default-table']['conditional']['border']['all']['color']
+			$json['componentStyles']['dark-table']['border']['all']['color']
 		);
 		$this->assertEquals(
-			'#fedcba',
-			$json['componentStyles']['default-table']['tableStyle']['cells']['conditional'][0]['backgroundColor']
+			'#abcdef',
+			$json['componentStyles']['dark-table']['tableStyle']['columns']['divider']['color']
 		);
 		$this->assertEquals(
-			'#123456',
-			$json['componentStyles']['default-table']['tableStyle']['cells']['conditional'][0]['textStyle']['textColor']
+			'#abcdef',
+			$json['componentStyles']['dark-table']['tableStyle']['rows']['divider']['color']
 		);
-
 		$this->assertEquals(
 			'#abcdef',
-			$json['componentStyles']['default-table']['tableStyle']['columns']['conditional'][0]['divider']['color']
+			$json['componentStyles']['dark-table']['tableStyle']['headerRows']['divider']['color']
 		);
+
+		// Ensure cell background and text colors match.
 		$this->assertEquals(
-			'#abcdef',
-			$json['componentStyles']['default-table']['tableStyle']['rows']['conditional'][0]['divider']['color']
+			'#fedcba',
+			$json['componentStyles']['dark-table']['tableStyle']['cells']['backgroundColor']
 		);
 		$this->assertEquals(
-			'#abcdef',
-			$json['componentStyles']['default-table']['tableStyle']['headerRows']['conditional'][0]['divider']['color']
+			'#123456',
+			$json['componentStyles']['dark-table']['tableStyle']['cells']['textStyle']['textColor']
 		);
 
+		// Ensure header background and text colors match.
 		$this->assertEquals(
 			'#654321',
-			$json['componentStyles']['default-table']['tableStyle']['headerCells']['conditional'][0]['backgroundColor']
+			$json['componentStyles']['dark-table']['tableStyle']['headerCells']['backgroundColor']
 		);
 		$this->assertEquals(
 			'#987654',
-			$json['componentStyles']['default-table']['tableStyle']['headerCells']['conditional'][0]['textStyle']['textColor']
+			$json['componentStyles']['dark-table']['tableStyle']['headerCells']['textStyle']['textColor']
+		);
+
+		// Test partial dark mode styles.
+		// Set table settings.
+		$this->set_theme_settings(
+			[
+				// Set default-table style.
+				'table_border_color'               => '#111111',
+				'table_body_background_color'      => '#000000',
+				// Reset dark mode style.
+				'table_body_background_color_dark' => '',
+			]
+		);
+
+		// Run the export.
+		$exporter = new Exporter( $content, $this->workspace, $this->settings );
```

Comment:
Ideally, this test would be updated to use the `get_json_for_post` method on the new testcase class. I've been updating tests as I've touched them. If it's straightforward to do, I'd recommend doing that vs. extending the old method.